cmd/xl-v1-metadata.go

```diff
@@ -50,7 +51,17 @@ func init() {
 	}
 	newPoly1305 := func(key []byte, mode bitrot.Mode) bitrot.Hash {
 		var pKey [32]byte
-		copy(pKey[:], key) // this is safe because bitrot.New will check the len of key
+
+		// If the r part of the key (key[:16]) is zero, the poly1305 multiply has no effect. It's
+		// very unlikely because the key will be randomly generated but we should prevent this.
+		// In this case me map the zero key to the inverse one (11111...).
+		if bytes.Equal(key[:16], pKey[:16]) {
```

Comment:
Okay maybe something like this?
```
Poly1305 uses the first 16 byte of the key for multiplication. If this part of the key is ever zero the multiply has no effect on the checksum. It is very unlikely that this will ever happen because the key is randomly generated, but even tough we should prevent this from happening. If the part of the key is zero, we now use the inverse - all bit's set. This will guarantee, that the multiplication effects the checksum. 
```